Source/cryptalgo/SecureSocketPort.h

```diff
@@ -72,19 +72,26 @@ namespace Crypto {
             };
 
         public:
+            // Type of underlying socket
+            enum class sockettype_t : uint8_t {
+                   CLIENT_SOCKET
+                ,  SERVER_SOCKET
+            };
+
             Handler(Handler&&) = delete;
             Handler(const Handler&) = delete;
             Handler& operator=(const Handler&) = delete;
```

Comment:
Please add the:
Handler& operator=(Handler&&) = delete;
for upgrading code we should add it now...